src/Features/LanguageServer/ProtocolUnitTests/Configuration/DidChangeConfigurationNotificationHandlerTest.cs

```diff
@@ -104,6 +105,48 @@ public class A { }";
             VerifyValuesInServer(server.TestWorkspace, clientCallbackTarget.MockClientSideValues);
         }
 
+        [Fact]
+        public void VerifyLspClientOptionNames()
+        {
+            var actualNames = DidChangeConfigurationNotificationHandler.SupportedOptions.Select(
+                DidChangeConfigurationNotificationHandler.GenerateFullNameForOption).ToArray();
+            // These options are persist in the LSP client. Please make sure also modify the LSP client code if these strings are changed.
+            var expectedNames = new[]
+            {
+                "symbol_search.dotnet_search_reference_assemblies",
+                "implement_type.dotnet_insertion_behavior",
+                "implement_type.dotnet_property_generation_behavior",
+                "completion.dotnet_show_name_completion_suggestions",
+                "completion.dotnet_provide_regex_completions",
```

Comment:
Shouldn't this be something like dotnet.completion.provideRegexCompletions? Or am I misunderstanding what this test is trying to do?